### ISSUE TITLE
How was the neg_sqr_dist formulated? Could you point out the equation in the paper? 

### ISSUE THREAD

**Issue by kalyanainala** [ChaofWang/L-GM_loss_pytorch#5]
In the python code L_GM_loss.py, the following lines were written
"
       XY = torch.matmul(feat, torch.transpose(self.means, 0, 1))
        XX = torch.sum(feat ** 2, dim=1, keepdim=True)
        YY = torch.sum(torch.transpose(self.means, 0, 1)**2, dim=0, keepdim=True)
        neg_sqr_dist = -0.5 * (XX - 2.0 * XY + YY)
"
I have read the paper and also looked at the original tensorflow code from 'https://github.com/WeitaoVan/L-GM-loss/blob/master/tensorflow/resnet_model.py' but couldn't understand how the neg_sqr_dist = -0.5 * (XX - 2.0 * XY + YY) equation was formed. Could you point out the equation in the paper which gave -0.5 * (XX - 2.0 * XY + YY) or explain it?

**Comment by kalyanainala** [comment]
I believe it is the equation 18 that gave the formula for neg_sqr_dist. ((a-b)**2 = (a**2 - 2ab + b**2)) where a, b are the xi and mean respectively.
![equation_18](https://user-images.githubusercontent.com/19593887/88145351-2513ee80-cbaf-11ea-8875-5e648080ddc5.JPG)
 If I am right, Why was the covariance matrix not considered in the equation?

**Comment by KAISER1997** [comment]
@kalyanainala Hi, have you tried implementing this after including the covariance matrix?
